recipe/meta.yaml

```diff
@@ -34,35 +31,42 @@ requirements:
     - {{ compiler('cxx') }}
     # llvm is needed for the headers
     - llvm-openmp              # [osx]
-    - patch     # [not win]
-    - m2-patch  # [win]
+
   host:
     - python
     - pip
     - setuptools
     - wheel
-    - llvmlite 0.38.*
+    - llvmlite 0.39.*
     - numpy
-    - tbb-devel 2021.*
-    - _openmp_mutex            # [linux and not ppc64le]
+    - tbb-devel 2021.*,<2021.6
+    - _openmp_mutex            # [linux]
+    - importlib_metadata       # [py<39]
+
   run:
     - python
-    - {{ pin_compatible('llvmlite', max_pin='x.x') }}
-    - {{ pin_compatible('numpy') }}
+    - importlib_metadata       # [py<39]
+    - llvmlite >=0.39.*,<1
+    # NumPy has a hard upper limit.
+    # NumPy 1.22.0, 1.22.1, 1.22.2 are all broken for ufuncs
+    # xref: https://github.com/numba/numba/issues/7756
+    # NumPy >=1.23 breaks with CUDA test: test_reinterpret_array_type
+    # xref: https://github.com/numba/numba/issues/8529
+    - numpy >=1.18,!=1.22.0,!=1.22.1,!=1.22.2,<1.23
     # needed for pkg_resources
     - setuptools
 
   run_constrained:
     - tbb  >=2021.0
```

Comment:
Should tbb upper bound be aligned with the host definition? tbb-devel 2021.*,<2021.6

Reply:
I don't think `tbb` and `tbb-devel` have the same pinnings here. The documentation covers the requirements of TBB in depth here: https://numba.readthedocs.io/en/0.56.3/user/installing.html

- **tbb-devel** - provides TBB headers/libraries for compiling TBB support into Numba’s threading backend (`2021 <=` version `< 2021.6` required).
- **tbb** - provides the TBB runtime libraries used by Numba’s TBB threading backend (version `>= 2021` required).

There is also a separate table that shows the same version requirements.

Reply:
Thanks, I thought I remembered they do something a bit odd with the version here.